integration_tests/src/main/python/hashing_test.py

```diff
@@ -31,36 +29,58 @@
     timestamp_gen,
     decimal_gen_32bit,
     decimal_gen_64bit,
-    decimal_gen_128bit]
-if not is_before_spark_320():
-    _xxhash_gens += [float_gen, double_gen]
+    decimal_gen_128bit,
+    float_gen,
+    double_gen
+]
 
 _struct_of_xxhash_gens = StructGen([(f"c{i}", g) for i, g in enumerate(_xxhash_gens)])
 
-_xxhash_fallback_gens = single_level_array_gens + nested_array_gens_sample + [
-    all_basic_struct_gen,
-    struct_array_gen,
-    _struct_of_xxhash_gens]
-if is_before_spark_320():
-    _xxhash_fallback_gens += [float_gen, double_gen]
+_xxhash_gens = (_xxhash_gens + [_struct_of_xxhash_gens] + single_level_array_gens
+                + nested_array_gens_sample + [
+                    all_basic_struct_gen,
+                    struct_array_gen,
+                    _struct_of_xxhash_gens
+                ] + map_gens_sample)
 
 @ignore_order(local=True)
 @pytest.mark.parametrize("gen", _xxhash_gens, ids=idfn)
 def test_xxhash64_single_column(gen):
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark : unary_op_df(spark, gen).selectExpr("a", "xxhash64(a)"))
+        lambda spark : unary_op_df(spark, gen).selectExpr("a", "xxhash64(a)"),
+        {"spark.sql.legacy.allowHashOnMapType" : True})
```

Comment:
What will happen if not set this conf? Does GPU align with Spark for this case ?

Reply:
>What will happen if not set this conf? 

Spark reports analysis error.

>Does GPU align with Spark for this case ?

Yes, also reports analysis error when Spark is in analysis phase.